handlers.c

```diff
@@ -795,27 +809,91 @@ bool handler__decinc(globals_t * vars, char **argv, unsigned argc)
     {
         m = (argc == 1) ? MATCHDECREASED : MATCHDECREASEDBY;
     }
-    else
+    else if (m == MATCHUNSPECIFIED)
     {
         show_error("unrecogised match type seen at decinc handler.\n");
         return false;
     }
 
+    bool rval = do_search ( m, vars, pval, argv, argc );
+
+    if ( m == MATCHRANGE )
+        free(pval);
+
+    return rval;
+}
+
+static bool try_get_range_values ( const char * arg, uservalue_t ** pval_out ) 
+{
+    char * argv_copy = strdupa(arg);
+
+    if (!argv_copy)
+        return false;
+
+    char * token = strtok(argv_copy, "..");
+
+    uservalue_t * pval = NULL;
+
+    if (pval_out)
+    {
+        pval = malloc(sizeof(uservalue_t)*2);
+        if ( !pval )
+            return false;
+    }
+
+    int tokens = 0;
+    while (token) 
+    {
+        bool retval;
+
+        if (pval_out)
+            retval = parse_uservalue_number(token, &pval[tokens]);
+        else
+        {
+            uservalue_t temp;
+            retval = parse_uservalue_number(token, &temp);
+        }
+        if (!retval)
+        {
+            free(pval);
+            show_error("bad value specified in range`");
+            return false;
+        }
+        token = strtok(NULL, "..");
+        tokens++;
+        if (tokens>1)
+            break;
+    }
+    if ( tokens > 1 )
+    {
+        if (pval_out)
+            *pval_out = pval;
+        return true;
+    }
+    else
+    {
+        free(pval);
+        return false;
+    }
+}
+
+static bool do_search ( scan_match_type_t m, globals_t * vars, uservalue_t * pval, char ** argv, unsigned argc ) 
+{
     if (vars->matches) {
-        if (checkmatches(vars, m, &val) == false) {
+        if (checkmatches(vars, m, pval) == false) {
             show_error("failed to search target address space.\n");
             return false;
         }
     } else {
         /* < > = != cannot be the initial scan */
-        if (argc == 1)
+        if (argc == 1 && m!=MATCHRANGE)
```

Comment:
Here the spaces are missing. This is hard to read and you've completely mixed styles.
This should be `m != MATCHRANGE`.